db/skipped_sequence.go

```diff
@@ -24,89 +24,127 @@ const (
 	DefaultClipCapacityHeadroom = 1000
 )
 
-type SkippedSequenceListEntry interface {
-	getTimestamp() int64
-	getLastSeq() uint64
-	setLastSeq(seq uint64)
-	getStartSeq() uint64
-	isRange() bool
-	getNumSequencesInEntry() int
-}
-
 // SkippedSequenceSlice stores the set of skipped sequences as an ordered slice of single skipped sequences
 // or skipped sequence ranges
 type SkippedSequenceSlice struct {
-	list                 []SkippedSequenceListEntry
+	list                 []*SkippedSequenceListEntry
 	ClipCapacityHeadroom int
 	lock                 sync.RWMutex
 }
 
-var _ SkippedSequenceListEntry = &SingleSkippedSequence{}
-
-// SingleSkippedSequence contains a single skipped sequence value + unix timestamp of the time it's created
-type SingleSkippedSequence struct {
-	seq       uint64
-	timestamp int64
+// SkippedSequenceListEntry contains start + end sequence for a range of skipped sequences +
+// a timestamp at which the entry was created in unix format. If an entry is a singular skipped sequence the end
+// sequence will not be 0
+type SkippedSequenceListEntry struct {
+	start     uint64 // start sequence of a range
+	end       uint64 // end sequence of the range (0 if a singular skipped sequence)
```

Comment:
It feels a bit off that the following are functionally equivalent, but have different handling based on isRange().
- start=n, end=0
- start=n, end=n
Are there benefits to the first approach, versus just using the second one every time we need a single skipped sequence?  It seems like we could avoid isRange checks in a number of places - maybe everywhere?

Reply:
Good point, think I tried to think of it in the same way as the previous design, will update to reflect your comment above.